homeassistant/components/tvoverlay/notify.py

```diff
@@ -0,0 +1,236 @@
+"""TvOverlay notification service for Android TV."""
+from __future__ import annotations
+
+import logging
+import os.path
+from typing import Any
+import uuid
+
+from tvoverlay import ImageUrlSource, Notifications
+from tvoverlay.const import (
+    COLOR_GREEN,
+    DEFAULT_APP_NAME,
+    DEFAULT_SMALL_ICON,
+    DEFAULT_SOURCE_NAME,
+    Positions,
+    Shapes,
+)
+
+from homeassistant.components.notify import (
+    ATTR_DATA,
+    ATTR_TITLE,
+    ATTR_TITLE_DEFAULT,
+    BaseNotificationService,
+)
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import (
+    ATTR_APP_ICON,
+    ATTR_APP_TITLE,
+    ATTR_BADGE_COLOR,
+    ATTR_BADGE_ICON,
+    ATTR_BG_COLOR,
+    ATTR_BORDER_COLOR,
+    ATTR_DURATION,
+    ATTR_ID,
+    ATTR_IMAGE,
+    ATTR_IMAGE_AUTH,
+    ATTR_IMAGE_ICON,
+    ATTR_IMAGE_PASSWORD,
+    ATTR_IMAGE_PATH,
+    ATTR_IMAGE_URL,
+    ATTR_IMAGE_USERNAME,
+    ATTR_MESSAGE_COLOR,
+    ATTR_PERSISTENT,
+    ATTR_POSITION,
+    ATTR_SHAPE,
+    ATTR_SOURCE_NAME,
+    ATTR_VIDEO,
+    ATTR_VISIBLE,
+    DEFAULT_DURATION,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_get_service(
+    hass: HomeAssistant,
+    config: ConfigType,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> TvOverlayNotificationService | None:
+    """Get the TvOverlay notification service."""
+    return (
+        TvOverlayNotificationService(discovery_info, hass.config.is_allowed_path)
```

Comment:
Is there a particular reason why you pass in `is_allowed_path`?

Otherwise this could be removed along with L78.

Reply:
Removed